src/components/PopupConfirm.js

```diff
@@ -0,0 +1,23 @@
+import Popup from "./Popup.js";
+import { selectors } from "../utils/constants.js";
+
+export default class PopupConfirm extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(selectors.popupForm);
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  setEventListeners() {
+    this._popupForm.addEventListener("submit", (event) => {
+      event.preventDefault();
+      this._handleFormSubmit();
+      this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form

Reply:
Thank you for the valuable feedback, I have corrected this.